### DISCUSSION TITLE
Webusb example on ESP32-S2

### DISCUSSION THREAD

**Discussion by xtrinch** [hathach/tinyusb#567] [General]
Noob question here:
Will the webusb example work with e.g. ESP32-S2-Saola-1, even though the esp-idf does not support USB host functionality? Or is the USB host functionality not required for webusb?

**Comment by ladyada** [comment]
you dont need USB host for webusb. however, we kinda recommend going with web serial instead of web usb because web usb is much harder to use.
https://codelabs.developers.google.com/codelabs/web-serial/?hl=ro#6

**Comment by xtrinch** [comment]
I'm wondering then, what makes the S2 a candidate for tinyusb support but not the good old esp32?
The ESP32-S2-Saola-1 board's specifications say that it includes a USB to UART bridge, is that not a problem?

**Comment by chegewara** [comment]
webusb, USB serial, USB hid etc are using native USB device, which old esp32 does not have. CP210x on saola board is not used in this case, you may use it only for serial logs or to flash, but all USB device implementations work on native USB which is using pins 19 and 20 on S2.

**Comment by ladyada** [comment]
you *can* use webusb with the S2....but we really recommend webserial for web-to-device interaction as it is so much easier to do *and* you can use your old esp32 :)

**Comment by chegewara** [comment]
My experience with webserial on ubuntu 20.04 is not good. I have to remove cdc_acm driver every time i connect esp32-S2 with this comand:
`sudo modprobe -r cdc_acm`
Working with webusb is much easier to me personally, but it is subjective of course.

**Comment by xtrinch** [comment]
Okay, thanks guys for answers & tips. I wish webserial was not an experimental feature of chrome, it'd make it more appealing.
